src/Nncase.Core/TensorUtilities.cs

```diff
@@ -233,9 +233,14 @@ public static int GetIndex(ReadOnlySpan<int> strides, ReadOnlySpan<int> indices,
     public static IR.Expr GetIndex(ReadOnlySpan<IR.Expr> strides, ReadOnlySpan<IR.Expr> indices, int startFromDimension = 0)
     {
         // Scalar
-        if (strides.Length == 0 || indices.Length == 0)
+        if (strides.Length == 0)
         {
-            throw new IndexOutOfRangeException();
+            if (indices.Length != 1 || indices[0] != (Expr)0)
```

Comment:
You can't check `indices[0] != 0` at compile time, it may not be a constant 0 but `(Expr)1 - 1`.
You can type:
```c#
if (indices.Length != 1 || indices[0] != (Expr)0)
{
    return new IndexOutOfRangeException();
}

return F.Math.Require(F.Math.NotEqual(indices[0], 0), 0);
```